e2e/validate/validation.go

```diff
@@ -181,3 +186,45 @@ func clustersInClusterSet(env *types.Env, config *types.Config, log *zap.Sugared
 
 	return nil
 }
+
+// detectDistro detects the cluster distribution. If distro is set in config, it uses the user
+// provided distro. Otherwise, it determines the distro, sets config.Distro and config.Namespaces.
+// Returns an error if distro detection fails or if clusters have inconsistent distributions.
+func detectDistro(env *types.Env, cfg *types.Config, log *zap.SugaredLogger) error {
+	if cfg.Distro != "" {
+		log.Infof("Using user selected kubernetes distribution: %q", cfg.Distro)
+
+		return nil
+	}
+
+	isHubOpenShift, err := IsOpenShiftCluster(env.Hub)
+	if err != nil {
+		return fmt.Errorf("failed to detect distro for cluster %q: %w", env.Hub.Name, err)
+	}
+
+	isC1OpenShift, err := IsOpenShiftCluster(env.C1)
+	if err != nil {
+		return fmt.Errorf("failed to detect distro for cluster %q: %w", env.C1.Name, err)
+	}
+
+	isC2OpenShift, err := IsOpenShiftCluster(env.C2)
+	if err != nil {
+		return fmt.Errorf("failed to detect distro for cluster %q: %w", env.C2.Name, err)
+	}
+
+	if isHubOpenShift != isC1OpenShift || isHubOpenShift != isC2OpenShift {
+		return fmt.Errorf("unsupported configuration, clusters have inconsistent distributions")
+	}
+
+	if isHubOpenShift {
+		cfg.Distro = config.DistroOcp
+		cfg.Namespaces = config.OcpNamespaces
+	} else {
+		cfg.Distro = config.DistroK8s
+		cfg.Namespaces = config.K8sNamespaces
+	}
+
+	log.Infof("Detected and set Kubernetes distribution: %q", cfg.Distro)
```

Comment:
Logging the namespaces can be helpful for new contributors. It makes it clear that we determined the namespaces at this point, and which namespaces we are going to use. Something like:

    log.Infof("Using namespaces %+v", config.Namespaces)

We can also add later, previously we did not log the namespaces when we selected them.